nil/services/synccommittee/proofprovider/task_handler.go

```diff
@@ -13,38 +15,60 @@ import (
 
 type HandlerTaskStorage interface {
 	AddTaskEntries(ctx context.Context, tasks ...*types.TaskEntry) error
+	GetActiveTaskCount(ctx context.Context) (int, error)
 }
 
 type taskHandler struct {
-	taskStorage HandlerTaskStorage
-	resultSaver TaskResultSaver
-	skipRate    int
-	taskNum     int
-	clock       clockwork.Clock
-	logger      logging.Logger
+	taskStorage    HandlerTaskStorage
+	resultSaver    TaskResultSaver
+	skipRate       int
+	taskNum        int
+	maxActiveTasks int
+	clock          clockwork.Clock
+	logger         logging.Logger
 }
 
 func newTaskHandler(
 	taskStorage HandlerTaskStorage,
 	resultSaver TaskResultSaver,
 	skipRate int,
+	maxActiveTasks int,
 	clock clockwork.Clock,
 	logger logging.Logger,
 ) api.TaskHandler {
 	return &taskHandler{
-		taskStorage: taskStorage,
-		resultSaver: resultSaver,
-		skipRate:    skipRate,
-		taskNum:     0,
-		clock:       clock,
-		logger:      logger,
+		taskStorage:    taskStorage,
+		resultSaver:    resultSaver,
+		skipRate:       skipRate,
+		maxActiveTasks: maxActiveTasks,
+		taskNum:        0,
+		clock:          clock,
+		logger:         logger,
 	}
 }
 
+func (h *taskHandler) IsReadyToHandle(ctx context.Context) (bool, error) {
+	active, err := h.taskStorage.GetActiveTaskCount(ctx)
+	if err != nil {
+		return false, err
+	}
+	return active <= h.maxActiveTasks, nil
```

Comment:
may be `<`, otherwise when `active == h.maxActiveTasks` will be handled max + 1 tasks